bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiBridgeHandler.java

```diff
@@ -0,0 +1,926 @@
+            webSocket.stop();
+            this.webSocket = null;
+        }
+    }
+
+    @Nullable
+    LivisiWebSocket createAndStartWebSocket(DeviceDTO bridgeDevice) throws IOException {
+        final Optional<String> accessToken = getAccessToken(client);
+        if (accessToken.isEmpty()) {
+            return null;
+        }
+
+        final String webSocketUrl = URLCreator.createEventsURL(bridgeConfiguration.host, accessToken.get(),
+                bridgeDevice.isClassicController());
+
+        logger.debug("WebSocket URL: {}...{}", webSocketUrl.substring(0, 70),
+                webSocketUrl.substring(webSocketUrl.length() - 10));
+
+        LivisiWebSocket webSocket = new LivisiWebSocket(httpClient, this, URI.create(webSocketUrl),
+                bridgeConfiguration.webSocketIdleTimeout * 1000);
+        webSocket.start();
+        return webSocket;
+    }
+
+    private static Optional<String> getAccessToken(LivisiClient client) throws IOException {
+        return Optional.of(client.getAccessTokenResponse().getAccessToken());
+    }
+
+    @Override
+    public void onAccessTokenResponse(final AccessTokenResponse credential) {
+        scheduleRestartClient(true);
+    }
+
+    /**
+     * Schedules a re-initialization in the given future.
+     *
+     * @param delayed when it is scheduled delayed, it starts with a delay of
+     *            {@link org.openhab.binding.livisismarthome.internal.LivisiBindingConstants#REINITIALIZE_DELAY_SECONDS}
+     *            seconds,
+     *            otherwise it starts directly
+     */
+    private synchronized void scheduleRestartClient(final boolean delayed) {
+        final ScheduledFuture<?> reInitJobLocal = this.reInitJob;
+        if (reInitJobLocal == null || !isAlreadyScheduled(reInitJobLocal)) {
+            long delaySeconds = 0;
+            if (delayed) {
+                delaySeconds = REINITIALIZE_DELAY_SECONDS;
+            }
+            logger.debug("Scheduling reinitialize in {} delaySeconds.", delaySeconds);
+            this.reInitJob = getScheduler().schedule(this::startClient, delaySeconds, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Starts a refresh job for the bridge channels, because the SHC 1 (classic) doesn't send events
+     * for cpu, memory, disc or operation state changes.
+     * The refresh job is only executed for SHC 1 (classic) bridges, newer bridges like SHC 2 do send events.
+     */
+    private void scheduleBridgeRefreshJob(DeviceDTO bridgeDevice) {
+        if (bridgeDevice.isClassicController()) {
+            final ScheduledFuture<?> bridgeRefreshJobLocal = this.bridgeRefreshJob;
+            if (bridgeRefreshJobLocal == null || !isAlreadyScheduled(bridgeRefreshJobLocal)) {
+                logger.debug("Scheduling bridge refresh job with an interval of {} seconds.", BRIDGE_REFRESH_SECONDS);
+
+                this.bridgeRefreshJob = getScheduler().scheduleAtFixedRate(() -> {
```

Comment:
SAT warning here, so just checking:

```
[WARNING] org.openhab.binding.livisismarthome.internal.handler.LivisiBridgeHandler.java:[300]
For periodically executed jobs that do not require a fixed rate scheduleWithFixedDelay should be preferred over scheduleAtFixedRate.
```

Is `scheduleAtFixedRate` really needed here?

Reply:
Yes, required. I don't know why this is a warning, delay and rate are totally different things. ;-)

Reply:
They are actually not that different, seems to be a matter of frequency accuracy: https://stackoverflow.com/questions/24649842/scheduleatfixedrate-vs-schedulewithfixeddelay

Reply:
Ok, I thought with delay is only executed once (and I thought I had already tested it ;-)). I have now changed it to delay, I don't think that it have an advantage or disadvantage in that case (the connection requires just a few seconds), but it solves the SAT warning...